beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -327,21 +327,49 @@ func validateAttestingIndex(
 }
 
 // Returns true if the attestation was already seen for the participating validator for the slot.
-func (s *Service) hasSeenCommitteeIndicesSlot(slot primitives.Slot, committeeID primitives.CommitteeIndex, aggregateBits []byte) bool {
+func (s *Service) hasSeenUnaggregatedAtt(att eth.Att) bool {
 	s.seenUnAggregatedAttestationLock.RLock()
 	defer s.seenUnAggregatedAttestationLock.RUnlock()
-	b := append(bytesutil.Bytes32(uint64(slot)), bytesutil.Bytes32(uint64(committeeID))...)
-	b = append(b, aggregateBits...)
+	var attester uint64
+	if att.Version() >= version.Electra {
+		if !att.IsSingle() {
+			log.Debug("Called hasSeenUnaggregatedAtt with a non-single Electra attestation")
+		}
+		attester = uint64(att.GetAttestingIndex())
+	} else {
+		aggBits := att.GetAggregationBits()
+		if aggBits.Count() != 1 {
+			log.Debug("Attestation does not have exactly 1 bit set. It will be considered not seen")
+			return false
+		}
+		attester = uint64(att.GetAggregationBits().BitIndices()[0])
+	}
+	b := append(bytesutil.Bytes32(uint64(att.GetData().Slot)), bytesutil.Bytes32(uint64(att.GetCommitteeIndex()))...)
+	b = append(b, bytesutil.Bytes32(attester)...)
```

Comment:
Optional feedback. We can be more performant here, Instead of appending multiple slices, preallocate a single slice and copy values in. Example:
```
// Preallocate 96 bytes (3 * 32 bytes) instead of multiple appends
b := make([]byte, 96)
binary.LittleEndian.PutUint64(b[:8], uint64(att.GetData().Slot))
binary.LittleEndian.PutUint64(b[32:40], uint64(att.GetCommitteeIndex()))
binary.LittleEndian.PutUint64(b[64:72], attester)
```